### DISCUSSION TITLE
metabat won't execute successfully

### DISCUSSION THREAD

**Discussion by ycsong** [franciscozorrilla/thesis_pipeline#155] [Q&A] [answered]
Hello,

I am able to submit the jobs to the node succesfully, but I am having some issues with metabat.

First a quick question. For metabat, I see that there needs to be two types of input:

* A gzipped fasta file named **contigs.fasta.gz** placed in the assemblies/<sample_code> folder.
* A 'cov' placed in metabat/<sample_code> folder

What is this 'cov' file, or is it a folder?  I am assuming that it is a coverage stats table from assemblies step that looks like:

```
#ID	Avg_fold	Length	Ref_GC	Covered_percent	Covered_bases	Plus_reads	Minus_reads	Read_GC	Median_fold	Std_Dev
CFS1-BTM1_c1	20.2561	2487	0.5690	99.3567	2471	166	170	0.5730	21	7.14
CFS1-BTM3_c1	22.2687	2021	0.6081	100.0000	2021	150	151	0.6106	24	9.39
CFS1-BTM4_c1	8.8488	2017	0.5786	100.0000	2017	58	60	0.5788	10	3.81
CFS1-BTM2_c1	31.9714	1995	0.5278	98.7970	1971	211	215	0.5032	42	21.06
```

It would be good if this could be confirmed.

My second question relates to setting up the conda environment. I saw it in the log that the following is trying to be executed:

`set +u;source activate envs/metagem;set -u`

I tried executing this code outside of metagem and then I got the following error message:

`EnvironmentLocationNotFound: Not a conda environment: /home/bioinfo_tools/bin/metaGEM/workflow/envs/metagem`

What would be a possible source of the issue?

Thank you in advance,

Young

**Comment by ycsong** [comment]
Update: I looked at this issue further and realized that I need to edit the "env" section of the config.yaml, such that whatever specified there is actually a conda environment. This appears to have solved the problem.

I do have a question: Is it possible to specify which data that I want to work with? For example, if I have subfolders A and B in the dataset folder, could I customize files or commands such that my run targets the B subfolder?

**Reply by franciscozorrilla** [reply, marked as answer]
> Update: I looked at this issue further and realized that I need to edit the "env" section of the config.yaml, such that whatever specified there is actually a conda environment. This appears to have solved the problem.

Glad you figured that one out!

> What is this 'cov' file, or is it a folder? I am assuming that it is a coverage stats table from assemblies step that looks like:

As I have recommended in our previous discussion, please have a look at the Snakefile rules to understand what is going on:
https://github.com/franciscozorrilla/metaGEM/blob/57cfd60485da82438faadcf75edc8623ac822196/workflow/Snakefile#L390-L492
In summary, this rule will generate the mapping information needed for each individual binner, and store it under the corresponding `binner/sample_ID/cov/` subfolder. Then, each binning job you submit will look into the appropriate folder in order to fetch the coverage information needed for binning :) 

I suggest you run the `crossMapSeries` rule above to see what the output files and folders look like. You may have to delete any empty `/cov` directories for the jobs to run, in case Snakemake didnt do that automatically when your last jobs failed.

> I do have a question: Is it possible to specify which data that I want to work with? For example, if I have subfolders A and B in the dataset folder, could I customize files or commands such that my run targets the B subfolder?

This is more of a general Snakemake question, so I recommend reading the [documentation](https://snakemake.readthedocs.io/en/stable/) for this. As a quick workaround, you could simply temporarily relocate the `/dataset` subfolders you are not interested out of that folder, then the Snakefile wont be able to find those samples to submit jobs.

**Comment by ycsong** [comment]
Thanks for your help Francisco. I think at this point, I can experiment with few things in Snakefile and other config documents to get things running (I've already did a couple of test runs in the server, and it seems to run well). If I do have more q's in the future, I'll post them in a separate discussion.

Thanks again,

Young
